### PR TITLE
RCHAIN-3265 add bonding test again

### PR DESCRIPTION
**Pull request by zsluedem**
## Overview
<sup>_What this PR does, and why it's needed_</sup>

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-3265

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Comment by zsluedem** [comment]
bors try

**Review by woky — APPROVED** [review]
Python looks good

**Comment by zsluedem** [comment]
@adaszko one more

File: integration-tests/test/test_heterogenous_validators.py [hunk 1]
```diff
@@ -1,6 +1,97 @@
...
+    wait_for_peers_count_at_least,
+    wait_for_blocks_count_at_least,
+)
+
+from .test_wallets import get_vault_balance
+
+BOOTSTRAP_KEY = PrivateKey.generate()
+BONDED_VALIDATOR_KEY = PrivateKey.generate()
+JOINING_VALIDATOR_KEY = PrivateKey.generate()
+READONLY_PEER_KEY = PrivateKey.generate()
```
Comment (adaszko) [inline integration-tests/test/test_heterogenous_validators.py]: It looks like this introduces a source of non-determinism in the test.  Non-determinism causes flaky tests.  We should avoid it.  Is it possible to seed the underlying random number generator?

**Review by adaszko — CHANGES_REQUESTED** [review]

**Review by adaszko — APPROVED** [review]

**Pull request merged**